lib/object/has-property.js

```diff
@@ -5,22 +5,39 @@
  * @author Eric Satterwhite
  **/
 
-module.exports = function hasOwnProperty(obj, prop) {
-  return Object.prototype.hasOwnProperty.call(obj, prop)
+module.exports = function hasOwnProperty(obj, string = '', sep = '.') {
+  /* eslint-disable-next-line no-eq-null */
+  if (obj == null) return false
+
+  const parts = string.split(sep)
+  let ret = obj
+  /* eslint-disable-next-line no-unused-vars */
+  const last = parts.pop()
+  let prop
+  while ((prop = parts.shift())) {
+    ret = ret[prop]
+    /* eslint-disable-next-line no-eq-null */
+    if (ret == null) return false
+  }
+
+  return Object.prototype.hasOwnProperty.call(ret, last)
```

Comment:
Curious if just using `object.get` is enough in most cases?

```javascript
const value = object.get('foo.bar.baz')
if (value == null) // its not set
```

Reply:
this was my thinking as well, although I'm not strongly opposed to the changes. I think in most cases though `get` is _probably_ enough even though there are some differences, e.g.:

```js
// this would be a case where the distinction is useful
const foo = {bar: {baz: 100}}
if (!object.has(foo, 'bar.toString')) // false
if (!object.get(foo, 'bar.toString')) // true
```

there is an argument that `get` should also call `has` too though.

Reply:
It feels too magical to me to mix the two.  For me, when this function is used, we're talking about visible own properties.  If I programmatically wanted to check a prototype method, I probably would just `Object.hasOwnProperty.call(object.get('some.thing.here'), 'toString')`

Reply:
> For me, when this function is used, we're talking about visible own properties

@darinspivey agreed, that's sort of what I'm getting at. `get` fits the same use-case for the most part, other than the fact that it does _not_ only look at visible own properties. `has` with a path sort of changes the semantics - you're no longer talking about visible own properties of the object you're passing in, but some other object down the path.

I guess my question is, does adding path support to `has` make more sense than using

```js
has(get('some.thing.here'), 'toString')
```
if you really are interested in a nested own-property check, vs just using `get` if you aren't?

Reply:
@mdeltito @darinspivey @esatterwhite how does it look now?